client/my-sites/upgrades/checkout-thank-you/failed-purchase.jsx

```diff
@@ -0,0 +1,71 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import Gridicon from 'components/gridicon';
+import { localize } from 'i18n-calypso';
+import PurchaseDetail from 'components/purchase-detail';
+
+const FailedPurchase = ( { failedPurchases, translate } ) => {
+	function getDescription() {
```

Comment:
Not sure if this is what you meant. This is a stateless functional component. Doesn't make much sense to me to extract to a method, but hey. :)